src/pages/home/report/ContextMenu/ContextMenuActions.js

```diff
@@ -264,6 +265,30 @@ export default [
         },
         getDescription: () => {},
     },
+    {
+        textTranslateKey: 'common.pin',
+        icon: Expensicons.Pin,
+        shouldShow: (type, reportAction, isArchivedRoom, betas, anchor, isChronosReport, reportID, isPinnedChat) => type === CONTEXT_MENU_TYPES.REPORT && !isPinnedChat,
+        onPress: (closePopover, {reportID}) => {
+            Report.togglePinnedState(reportID, false);
+            if (closePopover) {
+                hideContextMenu(false);
+            }
+        },
```

Comment:
Is it worth it to DRY this?

Reply:
You mean avoiding having two for pin/unpin? I tried a few options and even asked other engineers during ECX. Apparently, this is the way 🤷

Reply:
Yeah I noticed shouldShow is the same for both, and onPress is similar but with the opposite bool values, but I'm good with this as is